### DISCUSSION TITLE
Integration of NIST PQC standard algorithms

### DISCUSSION THREAD

**Discussion by bencemali** [openssl/openssl#25184] [Q&A] [answered]
If I understand correctly, openssl didn't start integrating pqc algorithms before as they were still under standardization. The final standards for [ML-KEM](https://csrc.nist.gov/pubs/fips/203/final), [ML-DSA](https://csrc.nist.gov/pubs/fips/204/final) and [SLH-DSA](https://csrc.nist.gov/pubs/fips/205/final) have been released.

- Does that mean that implementation/integration in openssl can start for these algorithms?
- If yes, what does the standard procedure look like and how long is it expected to last?
- Is there any chance that openssl would support hybrid combinations of pq and classical algorithms? Using these algorithms in hybrid combinations would be preferred in most cases for at least a few years.

**Comment by egyptiankarim** [comment]
Looks like OpenSSL partially addressed this question here: [Post-Quantum Algorithms in OpenSSL](https://openssl-corporation.org/post/2024-09-17-post-quantum/)

@baentsch any additional insight for us? Really excited to see this work progressing!

**Comment by baentsch** [comment, marked as answer]
Thanks for tagging me @egyptiankarim . Work on this proceeds in different feature branches, e.g., "ml-kem" or "slh-dsa". In any case, just search for the strings "ML-KEM", "MLKEM", "SLH-DSA" or "ML-DSA" to find open PRs and open issues, particularly if you'd like to lend a hand and thus speed up the work.

To answer the questions by @bencemali : "Yes", "by way of completion/merge of feature branches", and "[Yes](https://github.com/openssl/openssl/issues/25884)", respectively.
